types/fs-base-ui/html_ui/JS/SimPlane.d.ts

```diff
@@ -0,0 +1,421 @@
+        function getFMCApprGreenDotSpeed(): number | null;
+        function getGreenDotSpeed(): Knots | null;
+        function getCrossoverSpeedFactor(cas, mach: Mach): number | null;
+        function getFlapsLimitSpeed(aicraft: any, flapIndex: number): Knots | null;
+        /**
+         * @summary
+         * Seems to implement caching behaviour, can be overridden with forceSimVarCall.
+         */
+        function getFlapsHandleIndex(forceSimVarCall?: boolean): number | null;
+        /**
+         * @summary
+         * Seems to implement caching behaviour, can be overridden with forceSimVarCall.
+         */
+        function getFlapsExtendSpeed(aircraft: any, forceSimVarCall?: boolean): Knots | null;
+        /**
+         * @summary
+         * Seems to implement caching behaviour, can be overridden with forceSimVarCall.
+         */
+        function getNextFlapsExtendSpeed(aircraft: any, forceSimVarCall?: boolean): Knots | null;
+        function getMaxSpeed(aircraft: any): Knots;
+        function getLowestSelectableSpeed(): Knots | null;
+        function getStallProtectionMinSpeed(): Knots | null;
+        function getStallProtectionMaxSpeed(): Knots | null;
+        function getStallSpeed(): Knots | null;
+        function getStallSpeedPredicted(flapIndex: number): Knots | null;
+        function getWindDirection(): Degrees | null;
+        function getWindStrength(): Knots | null;
+        function getAutoPilotActive(apIndex: number): boolean | null;
+        function getAutoPilotAirspeedManaged(): boolean;
+        function getAutoPilotAirspeedSelected(): boolean;
+        function getAutoPilotAirspeedHoldActive(isManaged?: boolean): boolean | null;
+        function getAutoPilotAirspeedHoldValue(): Knots | null;
+        function getAutoPilotSelectedAirspeedHoldValue(): Knots | null;
+        function getAutoPilotManagedAirspeedHoldValue(): Knots | null;
+        function getAutoPilotMachModeActive(): boolean | null;
+        function getAutoPilotMachHoldValue(): number | null;
+        function getAutoPilotSelectedMachHoldValue(): number | null;
+        function getAutoPilotManagedMachHoldValue(): number | null;
+        function getAutoPilotHeadingManaged(): boolean;
+        function getAutoPilotHeadingSelected(): boolean;
+        function getAutoPilotHeadingLockActive(): boolean | null;
+        function getAutoPilotHeadingLockValue(radians?: boolean): Radians | Degrees | null;
+        function getAutoPilotSelectedHeadingLockValue(radians?: boolean): Radians | Degrees | null;
+        function getAutoPilotAltitudeManaged(): boolean;
+        function getAutoPilotAltitudeSelected(): boolean;
+        function getAutoPilotAltitudeArmed(): boolean | null;
+        function getAutoPilotAltitudeLockActive(): boolean | null;
+        function getAutoPilotFLCActive(): boolean;
+        /**
+         * @param units Default = feet.
+         */
+        function getAutoPilotAltitudeLockValue(units?: string): Feet | null;
```

Comment:
Could also return something other than feet if we change the unit. Maybe just say it returns `number | null`?